### PR TITLE
added check for _type_ value of request. added option responseType parameter.

### PR DESCRIPTION
**Pull request by iambumblehead**
added check for _type_ value of request. added option responseType parameter --used for blob and arraybuffer requests.

'blob' requests a development version of superagent at this time. no test added.

https://github.com/cyclejs/cycle-http-driver/issues/46

### CODE REVIEW

**Comment by iambumblehead** [comment]
I don't understand why the CI build fails. Tests are passing for me locally.

File: src/http-driver.js [hunk 1]
```diff
@@ -26,7 +27,12 @@ function optionsToSuperagent({
   if (typeof request.redirects === `function`) {
     request = request.redirects(redirects)
   }
-  request = request.type(type)
+  if (type) {
+    request = request.type(type)
+  }
+  if (responseType) {
+    request = request.reponseType(responseType)
```
Comment (staltz) [inline src/http-driver.js]: There's a typo there, `reponse` => `response`

**Comment by iambumblehead** [comment]
When I try running the full tests with `npm run test` I get this error

```bash
> @cycle/http@8.2.0 eslint /Users/chrisdep/Software/cycle-http-driver
> eslint src/

/Users/chrisdep/Software/cycle-http-driver/node_modules/estraverse/estraverse.js:517
                        throw new Error('Unknown node type ' + nodeType + '.');
                        ^

Error: Unknown node type AssignmentPattern.
    at Controller.traverse (/Users/chrisdep/Software/cycle-http-driver/node_modules/estraverse/estraverse.js:517:31)
    at EventEmitter.module.exports.api.verify (/Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/eslint.js:787:24)
    at processText (/Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/cli-engine.js:199:27)
    at processFile (/Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/cli-engine.js:224:12)
    at /Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/cli-engine.js:320:26
    at /Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/util/traverse.js:61:17
    at Array.forEach (native)
    at traverse (/Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/util/traverse.js:41:54)
    at walk (/Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/util/traverse.js:85:9)
    at /Users/chrisdep/Software/cycle-http-driver/node_modules/eslint/lib/util/traverse.js:104:9

npm ERR! Darwin 14.5.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "eslint"
npm ERR! node v5.5.0
npm ERR! npm  v3.7.0
npm ERR! code ELIFECYCLE
npm ERR! @cycle/http@8.2.0 eslint: `eslint src/`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @cycle/http@8.2.0 eslint script 'eslint src/'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @cycle/http package,
npm ERR! not with npm itself.
```

**Comment by iambumblehead** [comment]
I've cloned the latest 8.2.2 and am still unable to run the tests

**Comment by staltz** [comment]
Looking into it

**Comment by staltz** [comment]
Ok, the issue is with npm v3. Use npm v2 for now. It's not worth us spending time upgrading all these tools now, while we are rewriting to TypeScript and TSLint.

**Pull request closed without merging**